src/WebJobs.Script.WebHost/Diagnostics/AzureMonitorDiagnosticLogger.cs

```diff
@@ -93,6 +104,8 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
             using (JsonTextWriter writer = new JsonTextWriter(sw) { Formatting = Formatting.None })
             {
                 writer.WriteStartObject();
+                WritePropertyIfNotNull(writer, "appName", _appServiceOptions.CurrentValue.AppName);
```

Comment:
>_appServiceOptions.CurrentValue.AppName [](start = 58, length = 39)

Just double checking. The appName is different before and after specialization, right ? If yes, this does get refreshed after specialization ?

Reply:
Yes, it's registered with the Specialization change token: https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs#L146